neo/io/nwbio.py

```diff
@@ -0,0 +1,823 @@
+        "data", "starting_time_unit", "timestamps_unit", "electrode"
+    )
+
+    def __init__(self, timeseries, nwb_group):
+        self._timeseries = timeseries
+        self.units = timeseries.unit
+        if timeseries.conversion:
+            self.units = _recompose_unit(timeseries.unit, timeseries.conversion)
+        if timeseries.starting_time is not None:
+            self.t_start = timeseries.starting_time * pq.s
+        else:
+            self.t_start = timeseries.timestamps[0] * pq.s
+        if timeseries.rate:
+            self.sampling_rate = timeseries.rate * pq.Hz
+        else:
+            self.sampling_rate = None
+        self.name = timeseries.name
+        self.annotations = {"nwb_group": nwb_group}
+        self.description = try_json_field(timeseries.description)
+        if isinstance(self.description, dict):
+            self.annotations["notes"] = self.description
+            if "name" in self.annotations:
+                self.annotations.pop("name")
+            self.description = None
+        self.shape = self._timeseries.data.shape
+        if len(self.shape) == 1:
+            self.shape = (self.shape[0], 1)
+        metadata_fields = list(timeseries.__nwbfields__)
+        for field_name in self.__class__.common_metadata_fields:  # already handled
+            try:
+                metadata_fields.remove(field_name)
+            except ValueError:
+                pass
+        for field_name in metadata_fields:
+            value = getattr(timeseries, field_name)
+            if value is not None:
+                self.annotations[f"nwb:{field_name}"] = value
+        self.annotations["nwb_neurodata_type"] = (
+            timeseries.__class__.__module__,
+            timeseries.__class__.__name__
+        )
+        if hasattr(timeseries, "electrode"):
+            # todo: once the Group class is available, we could add electrode metadata
+            #       to a Group containing all signals that share that electrode
+            #       This would reduce the amount of redundancy (repeated metadata in every signal)
+            electrode_metadata = {"device": {}}
+            metadata_fields = list(timeseries.electrode.__class__.__nwbfields__) + ["name"]
+            metadata_fields.remove("device")  # needs special handling
+            for field_name in metadata_fields:
+                value = getattr(timeseries.electrode, field_name)
+                if value is not None:
+                    electrode_metadata[field_name] = value
+            for field_name in timeseries.electrode.device.__class__.__nwbfields__:
+                value = getattr(timeseries.electrode.device, field_name)
+                if value is not None:
+                    electrode_metadata["device"][field_name] = value
+            self.annotations["nwb_electrode"] = electrode_metadata
+
+    def load(self, time_slice=None, strict_slicing=True):
+        """
+        *Args*:
```

Comment:
I don't think this style of docstrings is used in many other classes of neo.. Maybe it would be good to use a more consistent style and check also that is is rendered properly in the documentation on readthedocs?